docs/examples/localstack.md

```diff
@@ -0,0 +1,142 @@
+
+    <!--codeinclude-->
+    [Localstack container running with a custom network](../../examples/localstack/localstack_test.go) inside_block:withNetwork
+    <!--/codeinclude-->
+
+* Other usage scenarios, such as where the Localstack container is used from both the test host and containers on a custom network are not automatically supported. If you have this use case, you should set `HOSTNAME_EXTERNAL` manually.
+
+## Module reference
+
+The LocalStack module exposes one single function to create containers, and this function receives three parameters:
+
+- `context.Context`
+- `OverrideContainerRequestOption`
+- a variadic argument of `LocalStackContainerOption`
+
+### OverrideContainerRequestOption
+
+The `OverrideContainerRequestOption` functional option represents a way to override the default LocalStack container request:
+
+<!--codeinclude-->
+[Default container request](../../examples/localstack/localstack.go) inside_block:defaultContainerRequest
+<!--/codeinclude-->
+
+With simply passing your own instance of an `OverrideContainerRequestOption` type to the `StartContainer` function, you'll be able to configure the LocalStack container with your own needs.
+
+In the following example you check how it's possible to set certain environment variables that are needed by the tests. Besides, the container runs in a separate Docker network with an alias:
+
+<!--codeinclude-->
+[Overriding the default container request](../../examples/localstack/localstack_test.go) inside_block:withNetwork
+<!--/codeinclude-->
+
+If you do not need to override the container request, you can pass `nil` or the `NoopOverrideContainerRequest` instance, which is exposed as a helper for this reason.
+
+<!--codeinclude-->
+[Skip overriding the default container request](../../examples/localstack/localstack_test.go) inside_block:noopOverrideContainerRequest
+<!--/codeinclude-->
+
+### LocalStackContainerOption, variadic argument
+
+#### WithLegacyMode
+
+The `WithLegacyMode` functional option represents a way to force LocalStack to run in legacy mode.
+
+<!--codeinclude-->
+[Forcing legacy mode](../../examples/localstack/localstack_legacy_mode_test.go) inside_block:forceLegacyMode
+<!--/codeinclude-->
+
+#### WithServices
+
+The `WithServices` functional option represents a way to pass as many AWS services as needed, in a variadic manner:
+
+<!--codeinclude-->
+[Passing AWS services to LocalStack](../../examples/localstack/v1/s3_test.go) inside_block:localStackCreateContainer
+<!--/codeinclude-->
+
+Using this method will internally populate the `SERVICES` environment variable, alongside the exposed ports of each service.
+
+### Available AWS Services
+
+The LocalStack module supports the following AWS services:
+
+- APIGateway
+- CloudFormation
+- CloudWatch
+- CloudWatchLogs
+- DynamoDB
+- DynamoDBStreams
+- EC2
+- Firehose
+- IAM
+- KMS
+- Kinesis
+- Lambda
+- Redshift
+- Route53
+- S3
+- SES
+- SNS
+- SQS
+- SSM
+- STS
+- SecretsManager
+- StepFunctions
```

Comment:
we should point to the docs instead of listing all services available by LocalStack. Otherwise, list need to be revisited to keep it up-to-date.

Reply:
Are all the services available in the Go/Java code? If not, we'll have to visit the code to update it when a new service comes. But in any case, it's safer to point to the docs. I'll change that ASAP